exchange/test/src/integration/basic.cpp

```diff
@@ -43,6 +43,37 @@ TEST_F(IntegrationBasicAlgo, InitialLiquidity)
     cycle.wait_for_order({buy, "TSLA", 10, 100});
 }
 
+TEST_F(IntegrationBasicAlgo, MarketOrderBuy)
+{
+    auto trader1 = start_wrappers(users_, "test_algos/buy_market_order_1000.py");
+    auto trader2 = users_.add_trader<TestTrader>(0);
+    trader2->modify_holdings("TSLA", 1000); // NOLINT
+    trader2->add_order({sell, "TSLA", 100, 1000});
+
+    TestMatchingCycle cycle{
+        {"TSLA"},
+        {trader1, trader2},
+    };
+
+    cycle.wait_for_order({buy, "APPL", 1, 1000});
```

Comment:
Shouldn't we be waiting for the market order? Why does buy_market_order place a limit order?